master/internal/command/shell_manager.go

```diff
@@ -52,12 +60,63 @@ func (s *shellManager) Receive(ctx *actor.Context) error {
 		}
 		ctx.Respond(resp)
 
+	case ShellLaunchRequest:
+		summary, err := s.processShellLaunchRequest(ctx, msg)
+		if err != nil {
+			ctx.Respond(errors.Wrap(err, "failed to launch shell"))
+		} else {
+			ctx.Respond(summary.ID)
+		}
+
 	case echo.Context:
 		s.handleAPIRequest(ctx, msg)
 	}
 	return nil
 }
 
+func (s *shellManager) processShellLaunchRequest(
+	ctx *actor.Context,
+	req ShellLaunchRequest,
+) (*summary, error) {
+	commandReq, err := parseCommandRequest(*req.User, s.db, req.CommandParams,
+		&s.taskSpec.TaskContainerDefaults,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if commandReq.AgentUserGroup == nil {
+		commandReq.AgentUserGroup = &s.defaultAgentUserGroup
+	}
+
+	var passphrase *string
+	if pwd, ok := commandReq.Data["passphrase"]; ok {
+		if typed, typedOK := pwd.(string); typedOK {
+			passphrase = &typed
+		}
+	}
+	generatedKeys, err := ssh.GenerateKey(passphrase)
```

Comment:
Go style nit: just use `key`, since this is such a short lived var. I know this stuff is a stupid comment, but for me it makes a big difference in readability (i'm a bad reader, sorry lol). I won't make this comment anymore :disappear:.

Reply:
lol no worries, shorter is easie!